src/libraries/System.IO.FileSystem/src/System/IO/Iterator.cs

```diff
@@ -14,13 +15,15 @@ internal abstract class Iterator<TSource> : IEnumerable<TSource>, IEnumerator<TS
     {
         private readonly int _threadId;
         internal int state;
-        internal TSource current;
+        [MaybeNull, AllowNull]
+        internal TSource current = default;
 
         public Iterator()
         {
             _threadId = Environment.CurrentManagedThreadId;
         }
 
+        [MaybeNull]
```

Comment:
Why is this needed?  Won't it only be null when TSource is non-nullable if the type is misused such that Current is accessed before the first MoveNext or after MoveNext has returned false?

Reply:
Yes that is the only case, don't need then?

Reply:
> don't need then?

Yeah, I think we can remove it.  Do we have [MaybeNull] on other `public T Current` properties elsewhere?  I want to make sure we're being consistent.

Reply:
I see one case have it https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Linq.Parallel/src/System/Linq/Parallel/Enumerables/EmptyEnumerable.cs#L68 will update that one too

Reply:
It's ok to update that one, but it's also internal and doesn't really matter.

Reply:
Updated for consistency 😉